hack/verify-canonical-import-path.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+staging_repos="${KUBE_ROOT}/staging/src"
+
+# verify-canonical-import path checks that files in the staging directory
+# have doc.go containing  canonical import paths.
+# See https://golang.org/doc/go1.4#canonicalimports for more details.
+
+failed=false
+for i in $(find $staging_repos -name 'doc.go'); do
```

Comment:
maybe it is not relevant here yet, but having `find` output in a for loop has size limits. Another way to express that:

```bash
while read i; do
  ...
done < <(find ...)
```

Reply:
Done.

> having find output in a for loop has size limits.

Didn't know this before, was an interesting read. :)